plutus-core/plutus-ir/src/PlutusIR/Transform/Inline/CallSiteInline.hs

```diff
@@ -154,7 +119,29 @@ fullyApplyAndBetaReduce info args0 = do
               (\n -> if n == param then Just <$> PLC.rename arg else pure Nothing)
               acc
           go acc' arity' args'
-        _ -> pure Nothing
+        -- term/type argument mismatch, don't inline
+        (TermParam _:_, TypeAppContext{}) -> pure Nothing
+        (TypeParam _:_, TermAppContext{}) -> pure Nothing
+        -- no more arguments to apply, just apply what we have
+        (remainingArity, AppContextEnd) ->
+          let numberOfUnappliedLam = length remainingArity
+              allLambdas = fst $ splitLambdas (varRhs info)
```

Comment:
This is not needed. The lambdas are already split - the binders are `varArity` and the body is `varBody`.

Reply:
The difference is that `Arity` doesn't have all the things needed to rebuild the term, e.g., for `Term`, it's missing `Type`, and for `Type`, it's missing `Kind`.

Reply:
Then we need to put additional things into `Arity`. We already have `splitParams`, so we don't need `splitLambdas`

Reply:
Yes I can alter `splitParams` instead.